src/coreclr/src/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -99,6 +102,42 @@ public Enumerator GetEnumerator()
                 return new Enumerator(_members);
             }
 
+            public IEnumerable<MethodDesc> MethodsRequiringSlotUnification
```

Comment:
Make this no allocation by doing this:

```suggestion
            public Enumerable MethodsRequiringSlotUnification => new Enumerable(_methodsRequiringSlotUnification);
```

With a matching:

```csharp
public readonly struct Enumerable
{
    private readonly MethodDesc[] _arrayToEnumerate;

    public Enumerable(MethodDesc[] arrayToEnumerate)
    {
        _arrayToEnumerate = arrayToEnumerate;
    }

    public Enumerator GetEnumerator()
    {
        return new Enumerator(_arrayToEnumerate);
    }
}
```